File: app/src/foss/java/com/simon/harmonichackernews/network/LocalSummaryManager.java
package com.simon.harmonichackernews.network;

import android.content.Context;

/** Fail-closed local summarization boundary for the FOSS distribution. */
final class LocalSummaryManager {
    private static final String UNAVAILABLE_MESSAGE =
            "Local AI is not included in the FOSS distribution.";

    private LocalSummaryManager() {
    }

    static boolean canAttemptLocalSummarization() {
        return false;
    }

    static void checkLocalSummaryAvailability(
            Context context, SummaryManager.LocalSummaryAvailabilityCallback callback) {
        SummaryManager.postLocalAvailability(callback, false, false, UNAVAILABLE_MESSAGE);
    }

    static void summarizeArticle(
            Context context, String articleUrl, SummaryManager.SummaryCallback callback) {
        SummaryManager.postFailure(callback, UNAVAILABLE_MESSAGE);
    }

    static void summarizeText(
            Context context, String text, SummaryManager.SummaryCallback callback) {
        SummaryManager.postFailure(callback, UNAVAILABLE_MESSAGE);
    }

    static boolean isLocalSummaryReady(Context context) {
        return false;
    }

    static boolean isLocalSummaryConfigurationKnown(Context context) {
        return true;
    }
}

File: app/src/foss/java/com/simon/harmonichackernews/network/LocalSummaryManager.kt
package com.simon.harmonichackernews.network

import android.content.Context
import com.simon.harmonichackernews.summary.StorySummaryBackend
import com.simon.harmonichackernews.summary.StorySummaryEvent
import com.simon.harmonichackernews.summary.StorySummaryInput
import com.simon.harmonichackernews.summary.LocalModelService
import com.simon.harmonichackernews.platform.LocalSummaryEngine
import com.simon.harmonichackernews.platform.PlatformCapability
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flowOf

internal class AndroidLocalSummaryBackend(context: Context) : StorySummaryBackend {
    override fun summarize(input: StorySummaryInput): Flow<StorySummaryEvent> = flowOf(
        StorySummaryEvent.Failure("Local AI is not included in the FOSS distribution."),
    )
}

internal fun createAndroidLocalSummaryCapability(
    context: Context,
    models: LocalModelService,
): PlatformCapability<LocalSummaryEngine> = PlatformCapability.Unavailable(
    name = "Local summaries",
    reason = "Local AI is not included in the FOSS distribution.",
)
